lms/djangoapps/certificates/api.py

```diff
@@ -273,6 +274,27 @@ def set_cert_generation_enabled(course_key, is_enabled):
         log.info(u"Disabled self-generated certificates for course '%s'.", unicode(course_key))
 
 
+def is_certificate_invalid(student, course_key):
+    """Check that whether the student in the course has been invalidated
+    for receiving certificates.
+
+    Arguments:
+        student (user object): logged-in user
+        course_key (CourseKey): The course identifier.
+
+    Returns:
+        Boolean denoting whether the student in the course is invalidated
+        to receive certificates
+    """
+    is_invalid = False
+    certificate = GeneratedCertificate.certificate_for_student(student, course_key)
+    if certificate and certificate.status == CertificateStatuses.unavailable:
+        # in case of status 'un-available', check that if this certificate is invalidated
```

Comment:
I read the code more and see what you are saying. Basically, invalidate action taken by the instructor would always result in the whichever certificate associated with the student for the course to be in `unavailable` status. So if the certificate is not in `unavailable` status, it always means the certificate was not invalidated.
That being said, what if there is a certificate that is not in `unavailable` status (say in `downloadable` status), but we have an `CertificateInvalidation` inserted for the student? In this case, should we return `false` in this function, or true?
To be on the safe side, I want this function to return `true`. But if you feel strongly that my supposed scenario above won't exist, I can be convinced as well. Maybe this is a product question? @marcotuts Do you have any opinions?